Automattic/pocket-casts-ios
# Fix Airplay stream issues

| 📘 Part of: #44 |  
|:---:|

Fixes #2429  


This PR ensures that when streaming/downloading a file that at the end of process we copy the file instead of moving to it does not interfere on the current streaming.
It also checks for the streaming cached file when starting the player to avoid downloading a file we already have.

## To test

1. Start the app
2. Ensure that you have the `streamAndCachePlayingEpisode` FF enabled
3. Start playing an episode that you didn't have download it before
4. Use Airplay and stream to your mac
5. Check if playback is working correctly
6. Switch from Airplay to phone speaker
7. Check if all is playing correct
8. Start playing another episode
9. While it's  playing press download on the episode detail
10. check if playing works correctly

## Checklist
